src/components/VmDetails/cards/NicsCard/NicListItem.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { msg } from '../../../../intl'
+import { escapeHtml } from '../../../utils'
+import { Grid, Row, Col } from '../../GridComponents'
+import style from './style.css'
+
+import { Icon, OverlayTrigger, Tooltip } from 'patternfly-react'
+import DeleteConfirmationModal from '../../../VmModals/DeleteConfirmationModal'
+import NicEditor from './NicEditor'
+import NicLinkStateIcon from './NicLinkStateIcon'
+
+/**
+ * Render a single NIC in the list of Nics on the Nics Card.
+ *
+ * If _edit_ then render the appropriate action buttons linked to provided
+ * handler functions.
+ */
+const NicListItem = ({ nic, vmStatus, vnicProfileList, isEditing, onEdit, onDelete }) => {
+  const canEdit = onEdit !== undefined
+  const canDelete = onDelete !== undefined
+
+  return <div className={style['nic-row']}>
+    {/* Status Column - content width only */}
+    <span className='nic-row-status'>
+      <NicLinkStateIcon linkState={nic.linked} idSuffix={nic.id} />
+    </span>
+
+    {/* Details Column - take the rest of the space */}
+    <span className={style['nic-row-info']}>
+      <div>
+        {nic.name}
+        { nic.vnicProfile.id
+          ? <span className={style['vnic-info']}>({nic.vnicProfile.name}/{nic.vnicProfile.network})</span>
+          : <span className={style['vnic-info']}>[{msg.nicNoVnicAssigned()}]</span>
+        }
+      </div>
+      <Grid>
+        <Row>
+          <Col cols={6} className={style['ip4-container']}>
+            { nic.ipv4.length > 0 &&
+              <div>
+                {nic.ipv4.map(ip4 => <div key={`${nic.id}-${ip4}`}>{msg.nicIP4()}: {ip4}</div>)}
+              </div>
+            }
+          </Col>
+          <Col cols={6} className={style['ip6-container']}>
+            { nic.ipv6.length > 0 &&
+              <div>
+                {nic.ipv6.map(ip6 => <div key={`${nic.id}-${ip6}`}>{msg.nicIP6()}: {ip6}</div>)}
+              </div>
+            }
+          </Col>
+        </Row>
+      </Grid>
+    </span>
+
+    {/* Actions Column (if edit) - content width, no wrapping */}
+    { isEditing &&
+    <span className={style['nic-row-actions']}>
+      { canEdit &&
```

Comment:
I'm actually using this style all over instead of ternary.  At this point, I'd rather maintain consistency with Utilization cards, Details card (#701), etc.

Reply:
Aha, now I see, and still don't understand, why to use two blocks instead of one logical block?

Reply:
> why to use two blocks instead of one logical block?

Because each one _is_ one logical block.  During development it helps me considerably to consider "what to display when the user CAN edit the field" to be separate and distinct from "what to display when the user CANNOT edit the field".  Keeping 2 blocks, even with inverse conditions, keeps the reasoning a bit less complex.

I know we have the ternary in lots of places for this kind of thing, but lately I've preferred to reserve ternary use for simple tags with a clear distinction (like the vnic label up towards line 34 in this file) and then use multiple blocks for complicated tags and the conditions are more complex.